src/KafkaFlow.Serializer/IMessageTypeResolver.cs

```diff
@@ -12,12 +13,12 @@ public interface IMessageTypeResolver
         /// </summary>
         /// <param name="context">The <see cref="IMessageContext"/> containing the message and the metadata</param>
         /// <returns></returns>
-        Type OnConsume(IMessageContext context);
+        Task<Type> OnConsumeAsync(IMessageContext context);
 
         /// <summary>
         /// Stores the message type somewhere when producing
         /// </summary>
         /// <param name="context">The <see cref="IMessageContext"/> containing the message and the metadata</param>
-        void OnProduce(IMessageContext context);
+        Task OnProduceAsync(IMessageContext context);
```

Comment:
Please use ValueTask as return type

Reply:
Is it because the implementations of this interface are synchronous?

Reply:
This is to reduce the heap allocations. The "hot path" of this interface implementation is sync, so it's better to use a ValueTask in these cases.